pulp_container/app/registry_api.py

```diff
@@ -689,34 +662,11 @@ def put(self, request, path, pk=None):
         _, repository = self.get_dr_push(request, path)
 
         digest = request.query_params["digest"]
-        # Try to see if the client came back after we told it to backoff with the ``Throttled``
-        # exception. In that case we answer based on the task state, or make it backoff again.
-        # This mechanism seems to work with podman but not with docker. However we let the task run
-        # anyway, since all clients will look with a HEAD request before attemting to upload a blob
-        # again.
         try:
             upload = models.Upload.objects.get(pk=pk, repository=repository)
         except models.Upload.DoesNotExist as e_upload:
-            # Upload has been deleted => task has started or even finished
-            task = Task.objects.filter(
-                name__endswith="add_and_remove",
-                reserved_resources_record__contains=[f"shared:upload:{pk}"],
-            ).last()
-            if not task:
-                # No upload and no task for it => the upload probably never existed
-                # return 404
-                raise e_upload
-
-            if task.state == "completed":
-                task.delete()
-                blob = models.Blob.objects.get(digest=digest)
-                return BlobResponse(blob, path, 201, request)
-            elif task.state in ["waiting", "running"]:
-                raise Throttled()
-            else:
-                error = task.error
-                task.delete()
-                raise Exception(str(error))
+            # return 404
+            raise e_upload
```

Comment:
Why reraising it now? I think the try block is not useful anymore now.

Reply:
Would calling https://docs.djangoproject.com/en/4.2/topics/http/shortcuts/#get-object-or-404 be more favourable?